### PR TITLE
[BBC] Introducing search by file type (tiff, png, jpeg)

### PR DESCRIPTION
**Pull request by gribeiro**
## What does this change?

This change introduces a new "advanced search" filter for searching by file type, named "fileType". 
The user can input some file formats like "jpg", "tif" (or "tiff"), "png" and the grid will search for the respective MIME type of the source image, i.e., "image/jpeg".

This change was introduced as part of https://github.com/bbc/grid/pull/55.

Edit: As per @paperboyo suggestion, we have added the offering of suggestions:

fileTypeSuggestions

## How can success be measured?

Upload file of types jpg, tiff, png and try to use the new filter: "fileType".

## Screenshots (if applicable)
searchByFileType

## Tested?
- [x] locally
- [x] on TEST

### CODE REVIEW

File: media-api/app/lib/querysyntax/QuerySyntax.scala [hunk 1]
```diff
@@ -178,6 +191,20 @@ class QuerySyntax(val input: ParserInput) extends Parser with ImageFields {
     })
   }
 
+  def MatchMimeTypeValue = rule {
+    (QuotedString | String) ~> parseMimeType _
+  }
+
+  def parseMimeType(expr: String): Value = {
```
Comment (akash1810) [inline media-api/app/lib/querysyntax/QuerySyntax.scala:198]: Could we make use of the [`MimeType` trait](https://github.com/guardian/grid/blob/e7d1d7f9f8263039c831b3c0c9255c6f0d318952/common-lib/src/main/scala/com/gu/mediaservice/model/MimeType.scala#L8) here to increase type safety?

Comment (gribeiro) [reply media-api/app/lib/querysyntax/QuerySyntax.scala]: Great idea. I also implemented a rule for just accepting the formats.

**Comment by paperboyo** [comment]
Great! From a user POV, since there is only a handful of fileformats supported, is there a way for it to offer suggestions like some other search chips?

image

**Comment by gribeiro** [comment]
Hello @paperboyo.
Nice suggestion. Just updated the PR with it. 😄

**Comment by akash1810** [comment]
This is looking great! Could we add a test too? [This PR](https://github.com/guardian/grid/pull/2190/files) may provide some inspiration.

**Comment by gribeiro** [comment]
Hello, @akash1810.

I've updated with the tests.
The build has some errors, but I think this is unrelated to my last commit. I had similar problem with my previous PR, Simon just ran it again and it works, maybe it is the same case?
If the build is ok, I'm good on the merging.

Thank you.

**Review by AWare — APPROVED** [review]
Works for me on TEST and looks fine to me 🚢

**Review by akash1810 — APPROVED** [review]

**Pull request merged**